packages/react/kibana_context/root/root_provider.tsx

```diff
@@ -6,10 +6,16 @@
  * Side Public License, v 1.
  */
 
-import type { I18nStart } from '@kbn/core-i18n-browser';
-import type { AnalyticsServiceStart } from '@kbn/core-analytics-browser';
 import React, { FC, PropsWithChildren } from 'react';
 
+import type { AnalyticsServiceStart } from '@kbn/core-analytics-browser';
+import type { I18nStart } from '@kbn/core-i18n-browser';
+
+// @ts-expect-error EUI exports this component internally, but Kibana isn't picking it up its types
+import { useIsNestedEuiProvider } from '@elastic/eui/lib/components/provider/nested';
+// @ts-expect-error EUI exports this component internally, but Kibana isn't picking it up its types
+import { emitEuiProviderWarning } from '@elastic/eui/lib/services/theme/warning';
```

Comment:
Did we look into adding this to kbn-ui-shared-deps?

Reply:
Thanks for the suggestion. I am looking into it now: dd3f12ac91b

Reply:
TIL this is an option - going to bookmark this for future reference 👀